test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -61,19 +63,36 @@ public static void main(String[] args) throws Exception {
 
         try {
             // create test directory
-            String tmpDir = System.getProperty("user.home");
-            testDir = new File[1];
-            testDir[0] = new File(tmpDir, "testDir");
-            if (!testDir[0].exists())
-                testDir[0].mkdir();
-            testDir[0].deleteOnExit();
-
-            // create temporary files inside testDir
-            tempFile = new File[5];
+            String tmpDir = System.getProperty("java.io.tmpdir");
+
+            // Create a test directory that contains only folders
+            testDir = new File(tmpDir, "testDir");
+            if (!testDir.exists()) {
+                testDir.mkdir();
+            }
+            testDir.deleteOnExit();
+
+            // create sub directories inside test directory
+            SubDirs = new File[5];
+            for (int i = 0; i < 5; ++i) {
+                SubDirs[i] = new File(testDir, "subDir_" + (i+1));
+                SubDirs[i].mkdir();
+                SubDirs[i].deleteOnExit();
+            }
+
+            // Create a test directory that contains only files
+            testFile = new File(tmpDir, "testFile");
+            if (!testFile.exists()) {
+                testFile.mkdir();
+            }
+            testFile.deleteOnExit();
```

Comment:
This is confusing: `testFile` is actually a directory.

You could improve the readability of the test by using methods: `createFoldersOnlyDir`, `createFilesOnlyDir`, `populateDirs`, `populateFiles` or something similar — it would make the comments in the code unnecessary.

Reply:
Test updated.